Sprint_5

Yana3473 opened the pull request:
Выполнила основные задания Sprint_5

Commit:
Выполнила основные задания Sprint_5

mkouznetsov-git commented on src/main/java/com/example/Lion.java:
```diff
@@ -4,29 +4,29 @@
+    private final Predator predator;
 
-    public Lion(String sex) throws Exception {
+    public Lion(String sex, Predator predator) throws Exception {
```
Вместо более общего интерфейса Predator следует использовать более точный класс Feline.

mkouznetsov-git commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Test
+    void testDoesHaveMane() {
```
В этом методе 2 разных проверки. Если первая упадёт, то вторая даже не будет выполнена. Раздели на 2 теста.

Commit:
В классе Lion использовала Predator predator, так как в задании был пункт: 3. Класс Lion не должен зависеть от класса Feline. Используй принцип инъекции зависимостей.

Сейчас исправила как вы сказали, но теперь как я поняла этот пункт не выполняется
